### PR TITLE
Warnint 64to32 6186 v23.2

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6186

Describe changes:
- fix `-Wshorten-64-to-32` warnings for some files : output, stream

Some commits of #9840
#11580 next batch
#11700 with code review taken into account

Still to do afterwards :
- detect
- util

Victor, I let you review the stream stuff ;-)

### CODE REVIEW

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on SURI_TLPR1_alerts_cmp.
  

Pipeline 22390

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 22430

File: src/output-json-http.c [hunk 1]
```diff
@@ -294,19 +295,12 @@ static void EveHttpLogJSONExtended(JsonBuilder *js, htp_tx_t *tx)
     const int resp = tx->response_status_number;
     if (resp > 0) {
         jb_set_uint(js, "status", (uint32_t)resp);
-    } else if (tx->response_status != NULL) {
```
Comment (victorjulien) [inline src/output-json-http.c:297]: This belongs in it's own commit

File: src/stream-tcp-reassemble.c [hunk 2]
```diff
@@ -1107,9 +1107,9 @@ static bool GetAppBuffer(const TcpStream *stream, const uint8_t **data, uint32_t
                     "got data at %"PRIu64". GAP of size %"PRIu64,
                     offset, blk->offset, blk->offset - offset);
             *data = NULL;
-            *data_len = blk->offset - offset;
+            *data_len = (uint32_t)(blk->offset - offset);
```
Comment (victorjulien) [inline src/stream-tcp-reassemble.c:1110]: not actually sure if we can't get a gap > 4GiB... It seems unlikely for sure. I guess we'd get all confused anyway due to sequence number wrapping around.

**Review by victorjulien — CHANGES_REQUESTED** [review]
Thanks for this cleanup. I have a few comments inline.

File: src/stream-tcp-reassemble.c (hunk 2 above)
Comment (catenacyber) [reply src/stream-tcp-reassemble.c:1110]: How could we tell that the GAP is over 4Gbytes with TCP ?

> I guess we'd get all confused anyway due to sequence number wrapping around.

Indeed

File: src/output-json-http.c (hunk 1 above)
Comment (catenacyber) [reply src/output-json-http.c:297]: and own issue and SV test with https://redmine.openinfosecfoundation.org/issues/7311

**Comment by catenacyber** [comment]
Next in https://github.com/OISF/suricata/pull/11904
